src/h5web/toolbar/Toolbar.tsx

```diff
@@ -24,48 +24,41 @@ function Toolbar(props: Props) {
   const childrenWidths = useMap<string, number>();
 
   // Group visible and hidden children based on their accumulated width
-  const [visibleChildren, hiddenChildren, allMeasured] = allChildren.reduce<
-    [ReactElement[], ReactElement[], boolean, number]
+  const [visibleChildren, hiddenChildren] = allChildren.reduce<
+    [ReactElement[], ReactElement[], number]
   >(
-    ([accVisible, accHidden, accAllMeasured, accWidth], child) => {
+    ([accVisible, accHidden, accWidth], child) => {
       const width = childrenWidths.get(child.key as string) ?? 0;
-      const isMeasured = width > 0;
       const isOverflowing = accWidth + width > availableWidth;
 
       return [
-        isMeasured && !isOverflowing ? [...accVisible, child] : accVisible,
-        isMeasured && isOverflowing ? [...accHidden, child] : accHidden,
-        accAllMeasured && isMeasured,
+        !isOverflowing ? [...accVisible, child] : accVisible,
+        isOverflowing ? [...accHidden, child] : accHidden,
         accWidth + width,
       ];
     },
-    [[], [], true, 0]
+    [[], [], 0]
   );
 
   const isSeparatorLast =
     visibleChildren[visibleChildren.length - 1]?.type === Separator;
 
-  const allOrVisibleChildren = allMeasured
-    ? isSeparatorLast
-      ? visibleChildren.slice(0, -1)
-      : visibleChildren
-    : allChildren;
```

Comment:
We no longer have to switch between rendering all the children or only the visible ones. We now assume that all children have a size of `0` to start, which means that they all end up in `visibleChildren` and they all get rendered and wrapped in a `MeasuredControl`.